From 3f115c58ea7736858b4ddd0c0035937e3aeb8508 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Wed, 15 Apr 2015 14:59:08 +0800
Subject: [PATCH] MLK-10573-2 dma: pxp-v2: add two planes output support

The output channel support two planes YUV formats.
So the output UV buffer should also be configured
in this case.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 drivers/dma/pxp/pxp_dma_v2.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/dma/pxp/pxp_dma_v2.c b/drivers/dma/pxp/pxp_dma_v2.c
index c850c5e..77c405e 100644
--- a/drivers/dma/pxp/pxp_dma_v2.c
+++ b/drivers/dma/pxp/pxp_dma_v2.c
@@ -407,6 +407,18 @@ static void pxp_set_outbuf(struct pxps *pxp)
 
 	__raw_writel(out_params->paddr, pxp->base + HW_PXP_OUT_BUF);
 
+	if ((out_params->pixel_fmt == PXP_PIX_FMT_NV12) ||
+		(out_params->pixel_fmt == PXP_PIX_FMT_NV21) ||
+		(out_params->pixel_fmt == PXP_PIX_FMT_NV16) ||
+		(out_params->pixel_fmt == PXP_PIX_FMT_NV61)) {
+		dma_addr_t Y, U;
+
+		Y = out_params->paddr;
+		U = Y + (out_params->width * out_params->height);
+
+		__raw_writel(U, pxp->base + HW_PXP_OUT_BUF2);
+	}
+
 	if (proc_data->rotate == 90 || proc_data->rotate == 270)
 		__raw_writel(BF_PXP_OUT_LRC_X(out_params->height - 1) |
 				BF_PXP_OUT_LRC_Y(out_params->width - 1),
-- 
1.8.0

